phpBB/phpbb/install/module/install_data/task/add_bots.php

```diff
@@ -0,0 +1,240 @@
+	 */
+	protected $io_handler;
+
+	/**
+	 * @var \phpbb\language\language
+	 */
+	protected $language;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string
+	 */
+	protected $php_ext;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\install\helper\config							$install_config		Installer's config
+	 * @param \phpbb\install\helper\iohandler\iohandler_interface	$iohandler			Input-output handler for the installer
+	 * @param \phpbb\install\helper\container_factory				$container			Installer's DI container
+	 * @param \phpbb\language\language								$language			Language provider
+	 * @param string												$phpbb_root_path	Relative path to phpBB root
+	 * @param string												$php_ext			PHP extension
+	 */
+	public function __construct(\phpbb\install\helper\config $install_config,
+								\phpbb\install\helper\iohandler\iohandler_interface $iohandler,
+								\phpbb\install\helper\container_factory $container,
+								\phpbb\language\language $language,
+								$phpbb_root_path,
+								$php_ext)
+	{
+		parent::__construct(true);
+
+		$this->db				= $container->get('dbal.conn');
+		$this->install_config	= $install_config;
+		$this->io_handler		= $iohandler;
+		$this->language			= $language;
+		$this->phpbb_root_path	= $phpbb_root_path;
+		$this->php_ext			= $php_ext;
+
+		/**
+		 * A list of the web-crawlers/bots we recognise by default
+		 *
+		 * Candidates but not included:
+		 * 'Accoona [Bot]'				'Accoona-AI-Agent/'
+		 * 'ASPseek [Crawler]'			'ASPseek/'
+		 * 'Boitho [Crawler]'			'boitho.com-dc/'
+		 * 'Bunnybot [Bot]'				'powered by www.buncat.de'
+		 * 'Cosmix [Bot]'				'cfetch/'
+		 * 'Crawler Search [Crawler]'	'.Crawler-Search.de'
+		 * 'Findexa [Crawler]'			'Findexa Crawler ('
+		 * 'GBSpider [Spider]'			'GBSpider v'
+		 * 'genie [Bot]'				'genieBot ('
+		 * 'Hogsearch [Bot]'			'oegp v. 1.3.0'
+		 * 'Insuranco [Bot]'			'InsurancoBot'
+		 * 'IRLbot [Bot]'				'http://irl.cs.tamu.edu/crawler'
+		 * 'ISC Systems [Bot]'			'ISC Systems iRc Search'
+		 * 'Jyxobot [Bot]'				'Jyxobot/'
+		 * 'Kraehe [Metasuche]'			'-DIE-KRAEHE- META-SEARCH-ENGINE/'
+		 * 'LinkWalker'					'LinkWalker'
+		 * 'MMSBot [Bot]'				'http://www.mmsweb.at/bot.html'
+		 * 'Naver [Bot]'				'nhnbot@naver.com)'
+		 * 'NetResearchServer'			'NetResearchServer/'
+		 * 'Nimble [Crawler]'			'NimbleCrawler'
+		 * 'Ocelli [Bot]'				'Ocelli/'
+		 * 'Onsearch [Bot]'				'onCHECK-Robot'
+		 * 'Orange [Spider]'			'OrangeSpider'
+		 * 'Sproose [Bot]'				'http://www.sproose.com/bot'
+		 * 'Susie [Sync]'				'!Susie (http://www.sync2it.com/susie)'
+		 * 'Tbot [Bot]'					'Tbot/'
+		 * 'Thumbshots [Capture]'		'thumbshots-de-Bot'
+		 * 'Vagabondo [Crawler]'		'http://webagent.wise-guys.nl/'
+		 * 'Walhello [Bot]'				'appie 1.1 (www.walhello.com)'
+		 * 'WissenOnline [Bot]'			'WissenOnline-Bot'
+		 * 'WWWeasel [Bot]'				'WWWeasel Robot v'
+		 * 'Xaldon [Spider]'			'Xaldon WebSpider'
+		 */
+		$this->bot_list = array(
```

Comment:
Same as above. Why are you defining this in the constructor and not by default for $this->bot_list?